src/photosui.cs

```diff
@@ -87,4 +91,232 @@ interface PHLivePhotoViewDelegate {
 		[Export ("livePhotoView:didEndPlaybackWithStyle:")]
 		void DidEndPlayback (PHLivePhotoView livePhotoView, PHLivePhotoViewPlaybackStyle playbackStyle);
 	}
+#else
+	[Mac (10,13, onlyOn64: true)][NoiOS][NoTV][NoWatch]
+	[Static]
+	interface PHProjectType {
+		[Field ("PHProjectTypeUndefined")]
+		NSString Undefined { get; }
+	}
+
+	[Mac (10,13, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSExtensionContext))]
+	interface PHProjectExtensionContext : NSSecureCoding, NSCopying {
+
+		[Export ("photoLibrary")]
+		PHPhotoLibrary PhotoLibrary { get; }
+
+		[Export ("project")]
+		PHProject Project { get; }
+	}
+
+	[Mac (10,13, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	[DisableDefaultCtor]
+	[BaseType (typeof (PHProjectElement))]
+	interface PHProjectJournalEntryElement : NSSecureCoding {
+
+		[Export ("date")]
+		NSDate Date { get; }
+
+		[NullAllowed, Export ("assetElement")]
+		PHProjectAssetElement AssetElement { get; }
+
+		[NullAllowed, Export ("textElement")]
+		PHProjectTextElement TextElement { get; }
+	}
+
+	[Mac (10,13, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	[DisableDefaultCtor]
+	[BaseType (typeof (PHProjectElement))]
+	interface PHProjectTextElement : NSSecureCoding {
+
+		[Export ("text")]
+		string Text { get; }
+
+		[NullAllowed, Export ("attributedText")]
+		NSAttributedString AttributedText { get; }
+
+		[Export ("textElementType")]
+		PHProjectTextElementType TextElementType { get; }
+	}
+
+	[Mac (10,13, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	[Protocol]
+	interface PHProjectExtensionController {
+
+		[Export ("supportedProjectTypes", ArgumentSemantic.Copy)]
+		PHProjectTypeDescription[] SupportedProjectTypes { get; }
```

Comment:
This should not be a property if is an `@optional` turn this into `GetSupportedProjectTypes()` or maybe is a `@required` one and is just missing `[Abstract]`?